### PR TITLE
Reuse operandType in checkPrefixUnaryExpression

### PR DESCRIPTION
**Pull request by jakebailey**
This was caught by jazzer as a timeout.

This code wasn't reusing `operandType`, so it was duplicating the check. If we nest this, this means that we're doing 2^n checks!

The test case here normally times out (and totally freezes tsserver; don't open it), but now finishes pretty much instantly.

### CODE REVIEW

**Comment by jakebailey** [comment]
@typescript-bot perf test this

**Comment by jakebailey** [comment]
Seems like no change. Let's test the rest.

@typescript-bot test this
@typescript-bot test top100
@typescript-bot user test this
@typescript-bot user test tsserver
@typescript-bot test tsserver top100
@typescript-bot run dt

File: tests/baselines/reference/nestedUnaryExpressionHang.errors.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+tests/cases/compiler/nestedUnaryExpressionHang.ts(1,6): error TS2363: The right-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type.
+tests/cases/compiler/nestedUnaryExpressionHang.ts(1,34): error TS1109: Expression expected.
+
+
+==== tests/cases/compiler/nestedUnaryExpressionHang.ts (2 errors) ====
+    3333%!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+!!! error TS2363: The right-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type.
```
Comment (jakebailey) [inline tests/baselines/reference/nestedUnaryExpressionHang.errors.txt:8]: This code doesn't parse properly, but, that's the test case that the fuzzer gave me so I feel like it's fine? But, I can try and come up with something that exhibits this problem without errors if desired.

File: src/compiler/checker.ts [hunk 2]
```diff
@@ -36120,7 +36120,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 }
                 return getUnaryResultType(operandType);
             case SyntaxKind.ExclamationToken:
-                checkTruthinessExpression(node.operand);
+                checkTruthinessOfType(operandType, node.operand);
```
Comment (jakebailey) [inline src/compiler/checker.ts:36123]: This direct call to `checkTruthinessOfType` when the operand type is known mirrors our handling of binary expressions in `onOperator`.

I checked all other call sites, and only this one benefits from not reevaluating the type.

**Review by andrewbranch — APPROVED** [review]

**Pull request merged**